GeneralsMD/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -5899,3 +6012,79 @@ void InGameUI::drawGameTime()
 	m_gameTimeString->draw(horizontalTimerOffset, m_gameTimePosition.y, m_gameTimeColor, m_gameTimeDropColor);
 	m_gameTimeFrameString->draw(horizontalFrameOffset, m_gameTimePosition.y, GameMakeColor(180,180,180,255), m_gameTimeDropColor);
 }
+
+void InGameUI::drawLatencyCounter(Int &x, Int &y)
+{
+	const UnsignedInt latencyFrames = TheNetwork->getRunAhead();
+
+	UnicodeString latencyStr;
+	latencyStr.format(L"%u", latencyFrames);
+	m_latencyString->setText(latencyStr);
```

Comment:
Text updates can be optimized by only refreshing them if the value has actually changed.